api/kv/value/value.go

```diff
@@ -130,6 +135,32 @@ func Uint(v uint) Value {
 	return Uint64(uint64(v))
 }
 
+// Array creates an ARRAY value.
+func Array(array interface{}) Value {
+	switch reflect.TypeOf(array).Kind() {
+	case reflect.Array, reflect.Slice:
+		isValidType := func() bool {
+			// get array type regardless of dimensions
+			typeName := reflect.TypeOf(array).String()
```

Comment:
Can we switch on `reflect.TypeOf(array).Elem().Kind()` instead of using string suffixes?

Reply:
`reflect.TypeOf(array).Elem().Kind()` works well in case of 1 dimensional array, but in case of multi dimensional array, it returns `reflect.Slice`, so have to iterate until get base type.

Reply:
OK. I still disagree with the need for a test at all, but I guess we can call that a spec-level disagreement. I will advocate for returning a valid ARRAY value if `array` is a `reflect.Slice` or `reflect.Array` or `nil`.